ev_charge/lib/widgets/electricity_prices_widget.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:ev_charge/core/get_electricity_prices.dart';
+import 'package:ev_charge/viewmodels/electricity_prices.dart';
+import 'package:http/http.dart' as http; // Import http
+
+class ElectricityPricesWidget extends StatefulWidget {
+  const ElectricityPricesWidget({super.key});
+
+  @override
+  State<ElectricityPricesWidget> createState() =>
+      _ElectricityPricesWidgetState();
+}
+
+class _ElectricityPricesWidgetState extends State<ElectricityPricesWidget> {
+  late Future<List<ElectricityPrices>> futureElectricityPrices;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final client = http.Client();
+    final service = ElectricityPricesService(client);
+    futureElectricityPrices = service.fetchElectricityPrices();
+  }
+
+  List<BarChartGroupData> getBarChartData(List<ElectricityPrices> prices) {
+    return prices.asMap().entries.map((entry) {
+      int index = entry.key;
+      ElectricityPrices data = entry.value;
+
+      return BarChartGroupData(
+        x: index,
+        barRods: [
+          BarChartRodData(
+            toY: data.price,
+            color: data.barColor,
+            width: 12,
+            borderRadius: BorderRadius.zero,
+          ),
+        ],
+      );
+    }).toList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<ElectricityPrices>>(
+      future: futureElectricityPrices,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(child: CircularProgressIndicator());
+        } else if (snapshot.hasError) {
+          return Center(child: Text('Error: ${snapshot.error}'));
+        } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+          return const Center(child: Text("No data available"));
+        }
+
+        final prices = snapshot.data!.reversed.toList();
+
+        return Column(
+          children: [
+            // Bar Chart
+            Container(
+              margin: const EdgeInsets.all(16.0),
+              width: MediaQuery.of(context).size.width * 0.9,
+              height: MediaQuery.of(context).size.height * 0.25,
+              child: BarChart(
+                BarChartData(
```

Comment:
```suggestion
                BarChartData(
                  barTouchData: BarTouchData(
                    enabled: true,
                    touchTooltipData: BarTouchTooltipData(
                      direction: TooltipDirection.top,
                      tooltipHorizontalAlignment: FLHorizontalAlignment.center,
                      fitInsideHorizontally: true,
                      fitInsideVertically: true,
                      tooltipPadding: EdgeInsets.all(8.0),
                      tooltipMargin: 0,
                    )
                  ),
```

Låser tooltipsne til at være inde i charten